tests/misc/t5870.nim

```diff
@@ -0,0 +1,14 @@
+discard """
+  file: "t5870.nim"
+  line: 11
+  errormsg: "invalid type for const: ref SomeRefObj"
+"""
+
+type SomeRefObj = ref object of RootObj
+  someIntMember: int
+
+proc createSomeRefObj(v: int): SomeRefObj =
+  result.new()
```

Comment:
Never seen initialization done like this :o

Reply:
@dom96 well I just copy-pasted this from issue

Reply:
It's amazing how many different ways to do the same thing there are. No wonder there are so many bugs :)